### DISCUSSION TITLE
`opendcs lite`

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#264] [Ideas]
How difficult would it be to separate out only those pieces that are required for `getDcpMessages` and remove everything else? Ideally, if someone implemented just the `getDcpMessages` in Python, that is all I need.  I tried navigating the code but as I am unfamiliar with Java, I couldn't figure out much. For someone like me who only needs a way to retrieve messages, this would be very helpful. I could figure out alternative ways to handle the scheduling and processing of messages.

**Comment by MikeNeilson** [comment]
Separating out the Java code, a royal pain. However if you really just want to handle the `getDcpMessages` functionality the communications protocol to the LRGS is based on a standard.

https://github.com/opendcs/dcs_standards/blob/main/source/dds-14.rst should get you started if you want to implement a client in python.
